content/docs/aro/blob-storage-csi/index.md

```diff
@@ -0,0 +1,315 @@
+
+1. In the Azure Portal, go to AAD and register a new application with a proper name to identify that this registration is for the driver to access the Blob Storage
+
+![Image](Images/blob-sp0.png)
+
+![Image](Images/blob-sp1.png)
+
+1. Once created, take note of the Client Id in the overview section.
+
+
+![Image](Images/blob-sp1-1.png)
+
+1. Set an environment variable with the value of the Client ID.
+
+    ```bash
+    export AAD_CLIENT_ID=REPLACE-WITH-YOUR-CLIENT-ID
+    ```
+
+1. Then, you need to create a secret for this registration. Go to "certificates & secrets" within the registration and create the new secret.
+
+![Image](Images/blob-sp2.png)
+
+1. Take note of the secret just created, and copy the value of it.
+
+![Image](Images/blob-sp3.png)
+
+1. Set another environment variable with the value of the Client Secret Value.
+
+    ```bash
+    export AAD_CLIENT_SECRET=REPLACE-WITH-YOUR-CLIENT-SECRET-VALUE
+    ```
+
+1. Set other environment variables:
+
+    > Update the `LOCATION`, `CLUSTER_NAME`, and `RG_NAME` variables to match your cluster details
+
+    ```bash
+    export LOCATION=eastus
+    export CLUSTER_NAME=my-cluster
+    export RG_NAME=myresourcegroup 
+    export TENANT_ID=$(az account show --query tenantId -o tsv)
+    export SUB_ID=$(az account show --query id)
+    export MANAGED_RG=$(az aro show -n $CLUSTER_NAME -g $RG_NAME --query 'clusterProfile.resourceGroupId' -o tsv)
+    export MANAGED_RG_NAME=`echo -e $MANAGED_RG | cut -d  "/" -f5`
+    ```
+
+1. Once we have all the environment variables set, we need to create the configuration file we will use to populate a json structure with all the data needed. 
+
+
+    ```bash
+    cat <<EOF >> cloud.conf
+    {
+    "tenantId": "$TENANT_ID",
+    "subscriptionId": $SUB_ID,
+    "resourceGroup": "$MANAGED_RG_NAME",
+    "location": "$LOCATION",
+    "useManagedIdentityExtension": false,
+    "aadClientId": "$AAD_CLIENT_ID",
+    "aadClientSecret": "$AAD_CLIENT_SECRET"
+    }
+    EOF
+    ```
+
+1. Check that all the attributes are populated. 
+
+>NOTE: Take care when executing this validation, since sensitive information should be disclosed in your screen. 
+
+   ```bash
+   cat cloud.conf
+   ```
+
+1. Create the project where you are going to install the driver.
+
+    ```bash
+    export CSI_BLOB_PROJECT=csi-azure-blob
+    oc new-project ${CSI_BLOB_PROJECT}
+    ```
+
+1. Then, create the secret into the project just created.
+
+    ```bash
+    export CSI_BLOB_SECRET=csi-azure-blob-secret
+    oc create secret generic ${CSI_BLOB_SECRET} --from-file=cloud-config=cloud.conf
+    ```
+
+
+## Driver installation
+
+Now, we need to install the driver, which could be done using a helm chart. 
+
+1. Assign permissions to the service accounts defined in the helm chart for the driver pods.
+
+    ```bash 
+    cat <<EOF | oc apply -f -
+    allowHostPorts: true
```

Comment:
write this resource properly so it shows the apiversion and kind first.

Reply:
done